Index: webrtc/base/openssladapter.cc
===================================================================
--- webrtc/base/openssladapter.cc	(revision 8096)
+++ webrtc/base/openssladapter.cc	(working copy)
@@ -878,6 +878,9 @@
   SSL_CTX_set_info_callback(ctx, SSLInfoCallback);
 #endif
 
+  // This fixes DTLS handshake. Without this option partial reads discard unread UDP bytes.
+  SSL_CTX_set_read_ahead(ctx, 1);
+
   SSL_CTX_set_verify(ctx, SSL_VERIFY_PEER, SSLVerifyCallback);
   SSL_CTX_set_verify_depth(ctx, 4);
   SSL_CTX_set_cipher_list(ctx, "ALL:!ADH:!LOW:!EXP:!MD5:@STRENGTH");
Index: webrtc/base/opensslstreamadapter.cc
===================================================================
--- webrtc/base/opensslstreamadapter.cc	(revision 8096)
+++ webrtc/base/opensslstreamadapter.cc	(working copy)
@@ -758,6 +758,9 @@
     mode |= SSL_VERIFY_FAIL_IF_NO_PEER_CERT;
   }
 
+  // This fixes DTLS handshake. Without this option partial reads discard unread UDP bytes.
+  SSL_CTX_set_read_ahead(ctx, 1);
+
   SSL_CTX_set_verify(ctx, mode, SSLVerifyCallback);
   SSL_CTX_set_verify_depth(ctx, 4);
   SSL_CTX_set_cipher_list(ctx, "ALL:!ADH:!LOW:!EXP:!MD5:@STRENGTH");
